### DISCUSSION TITLE
Architecture Verification: Consuming Web Services with .NET MAUI

### DISCUSSION THREAD

**Discussion by RafikMk** [dotnet/maui#15430] [General]
I have designed an architecture for my .NET MAUI application that involves consuming web services. I want to ensure that I'm following best practices and making the most efficient use of the framework. Here are the key components of my architecture:
![image](https://github.com/dotnet/maui/assets/61156835/bead5b6f-04b9-4f9b-9b82-1b6288a4682e)

**Presentation Layer (.NET MAUI App):** This layer includes the View, ViewModel, and Service folders.

- The View folder contains the classes responsible for the user interface of your application.

- The ViewModel folder contains the classes that manage the business logic and communication between the presentation layer and the service layer.

- The Service folder contains custom service classes used for communication with web services.

**Business Layer (.NET Framework):** This layer interacts with the persistence layer to access and manipulate data. It includes:

- The web service developed with the .NET Framework.

- The web service generates WSDL files that describe how it can be used.

**Persistence Layer (SQL Server):** This layer represents the SQL Server database used to store application data. The web service communicates with the database to read and write data.

I would like to get ideas and recommendations on this architecture. Are there any potential issues or improvements that I should consider? Are there any best practices for consuming web services in a .NET MAUI architecture that I might be missing?

**Comment by JaimeBernar** [comment]
@RafikMk Even it seems like a good archiecture I recommend you to check https://github.com/jasontaylordev/CleanArchitecture. This will decouple more the View form the logic so in the future you can use another UI framework. For what I can see you will have all the models, logic and UI in the same layer wich is not a good practice. Check the link and let me know what do you think.
